Xamarin.Forms.Platform.Android/Forms.cs

```diff
@@ -191,54 +193,149 @@ static Color GetAccentColor()
+				SetScreenOrientation();
+				_formsActivity.ConfigurationChanged += OnConfigurationChanged;
 			}
 
-			public override Size PixelScreenSize
+			public override Size PixelScreenSize => _pixelScreenSize;
+
+			public override Size ScaledScreenSize { get; }
+
+			public override double ScalingFactor => _scalingFactor;
+
+			public override void BeginDeviceOrientationNotifications()
 			{
-				get { return _pixelScreenSize; }
+				if (_androidOrientationEventListener != null)
+					return;
+
+				_androidOrientationEventListener = new AndroidOrientationEventListener((Context)_formsActivity, SensorDelay.Normal);
+				_androidOrientationEventListener.OrientationChanged += OnDeviceOrientationChanged;
+
+				if (_androidOrientationEventListener.CanDetectOrientation())
+					_androidOrientationEventListener.Enable();
 			}
 
-			public override Size ScaledScreenSize { get; }
+			public override void EndDeviceOrientationNotifications()
+			{
+				if (_androidOrientationEventListener == null)
+				    return;
+
+				_androidOrientationEventListener.OrientationChanged -= OnDeviceOrientationChanged;
+				_androidOrientationEventListener.Disable();
+				_androidOrientationEventListener.Dispose();
+				_androidOrientationEventListener = null;
+			}
+
+			void OnDeviceOrientationChanged(object sender, int i)
+			{
+				SetDeviceOrientation(i);
+			}
+
+			void OnConfigurationChanged(object sender, EventArgs e)
+			{
+				SetScreenOrientation();
+			}
 
-			public override double ScalingFactor
+			void SetDeviceOrientation(int rotation)
 			{
-				get { return _scalingFactor; }
+				if (rotation == OrientationEventListener.OrientationUnknown)
+				{
+					DeviceOrientation = DeviceOrientation.Unknown;
+					return;
+				}
+
+				if (PixelScreenSize.Width < PixelScreenSize.Height)
+				{
+					const int threshold = 45;
+
+					if (rotation <= threshold || rotation > 360 - threshold)
+						DeviceOrientation = DeviceOrientation.Portrait;
+					else if (rotation <= 360 - threshold && rotation > 270 - threshold)
+						DeviceOrientation = DeviceOrientation.Landscape;
+					else if (rotation <= 270 - threshold && rotation > 180 - threshold)
+						DeviceOrientation = DeviceOrientation.PortraitFlipped;
+					else if (rotation <= 180 - threshold && rotation > threshold)
+						DeviceOrientation = DeviceOrientation.LandscapeFlipped;
+				}
+				else if (PixelScreenSize.Width > PixelScreenSize.Height)
+				{
+					const int threshold = 45;
+
+					if (rotation <= threshold || rotation > 360 - threshold)
+						DeviceOrientation = DeviceOrientation.Landscape;
+					else if (rotation <= 360 - threshold && rotation > 270 - threshold)
+						DeviceOrientation = DeviceOrientation.PortraitFlipped;
+					else if (rotation <= 270 - threshold && rotation > 180 - threshold)
+						DeviceOrientation = DeviceOrientation.LandscapeFlipped;
+					else if (rotation <= 180 - threshold && rotation > threshold)
+						DeviceOrientation = DeviceOrientation.Portrait;
+				}
```

Comment:
I haven't fully read [this](http://www.thedroidsonroids.com/blog/android/simple-multi-window-part-1-screen-changes/), but it seems to be claimed that orientation changes are still detected. I will test multi window mode on Android tonight and get back. 

As for iOS, I don't have an iPad, so I can't test slide over or split view at the moment.